### PR TITLE
chore: automate the release process

### LINKED ISSUE #983: Add GitHub Action to automate release process
## Overview
The manual release process for our repository, now housing over 20 crates, is proving to be inefficient. To enhance workflow and ensure crates are published in the correct order, it's essential to automate the release process.

### PR DESCRIPTION
**Pull request by Farhad-Shabani**
Closes: #983
______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [X] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [X] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# release.sh will hopefully allow us to publish all of the necessary crates in
+# this repo in the right order. It is assumed that only one person will be
+# releasing all crates at the same time.
+
+set -e
+
+# A space-separated list of all the crates we want to publish, in the order in
+# which they must be published. It's important to respect this order, since
+# each subsequent crate depends on one or more of the preceding ones.
+DEFAULT_CRATES="ibc-primitives \
```
Comment (rnbguy) [inline scripts/release.sh]: can we use an array here?

File: scripts/release.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+# Allows us to override the crates we want to publish.
+CRATES=${*:-${DEFAULT_CRATES}}
+
+# Additional flags to pass to the "cargo publish" operation for every crate we
+# publish.
+CARGO_PUBLISH_FLAGS=""
+
+# Allow us to specify a crates.io API token via environment variables. Mostly
+# for CI use.
+if [ -n "${CRATES_TOKEN}" ]; then
```
Comment (rnbguy) [inline scripts/release.sh]: I suggest to make [empty string comparisons](https://github.com/anordal/shellharden/blob/master/how_to_do_things_safely_in_bash.md#are-empty-string-comparisons-any-special) to generate `"${var}" != ""`.

File: scripts/release.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+# release.sh will hopefully allow us to publish all of the necessary crates in
+# this repo in the right order. It is assumed that only one person will be
+# releasing all crates at the same time.
+
+set -e
```
Comment (rnbguy) [inline scripts/release.sh]: you should turn on `set -o pipefail` too.

**Review by rnbguy — CHANGES_REQUESTED** [review]
Thanks @Farhad-Shabani :pray:

File: scripts/release.sh (hunk 3 above)
Comment (rnbguy) [reply scripts/release.sh]: and, `set -u` too.

[ref](https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425)

Comment (Farhad-Shabani) [reply scripts/release.sh]: [fcdbd93](https://github.com/cosmos/ibc-rs/pull/984/commits/fcdbd93cfd6c43fd2a5ed86a1477a49f4ad27674)

File: scripts/release.sh (hunk 2 above)
Comment (Farhad-Shabani) [reply scripts/release.sh]: [fcdbd93](https://github.com/cosmos/ibc-rs/pull/984/commits/fcdbd93cfd6c43fd2a5ed86a1477a49f4ad27674)

File: scripts/release.sh (hunk 1 above)
Comment (Farhad-Shabani) [reply scripts/release.sh]: [fcdbd93](https://github.com/cosmos/ibc-rs/pull/984/commits/fcdbd93cfd6c43fd2a5ed86a1477a49f4ad27674)

**Review by rnbguy — APPROVED** [review]
Thanks, Farhad, for taking care of the automated release. :clap: 

I added two extra commits.

**Comment by Farhad-Shabani** [comment]
Awesome, thanks for the adjustments @rnbguy 👌🏻

**Pull request merged**